### PR TITLE
Splitting jsonpatches on a conditional antrea installation for windows

### LINKED ISSUE #3866: Enable the CNI:none option on Windows nodes
**Describe the feature request**
CNI: none is ignored for Windows nodes, it is required to work in order for custom CNI installations work in the cp/nodes.

**Describe alternative(s) you've considered**
None

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- ( ) Addons
- ( ) CLI
- ( ) Docs
- ( ) IAM
- (x) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- (x) User Experience
- ( ) Developer Experience

**Additional context**

### PR DESCRIPTION
**Pull request by knabben**
### What this PR does / why we need it
Splitting conditional JSON patches for Antrea and Windows installations, this will allow operator to use CNI:none and do not have antrea by default in the windows node.

### Which issue(s) this PR fixes

Fixes https://github.com/vmware-tanzu/tanzu-framework/issues/3866

### Describe testing done for PR
Manual tests with the CC were made

### Release note

```release-note
CNI:none enabled on ClusterClass Windows nodes.
```

### CODE REVIEW

**Comment by alfredthenarwhal** [comment]
Tests failed! Build no: 3158

**Comment by vuil** [comment]
cc @tenczar to take a look.
@knabben can you add what testing was done with this change to the PR description. It seems at minimum we need to test the effect of CNI:none on windows and non-windows nodes.

Could you also have demonstrated the effect of these changes using clustergen?

**Review by tenczar — APPROVED** [review]
LGTM, One change, please bump the annotation at line 174 of base.yml `identifier.cc.vmware.com: tanzu-vsphere-default-0.0.2` this version is used by the UI team to track if changes have been made to the variable schema.

File: packages/tkg-clusterclass-vsphere/bundle/config/upstream/base.yaml [hunk 1]
```diff
@@ -171,7 +171,7 @@ metadata:
   namespace: #@ data.values.NAMESPACE
   annotations:
     run.tanzu.vmware.com/resolve-tkr: ""
-    identifier.cc.vmware.com: tanzu-vsphere-default-0.0.2
+    identifier.cc.vmware.com: tanzu-vsphere-default-0.0.3
```
Comment (wjun) [inline packages/tkg-clusterclass-vsphere/bundle/config/upstream/base.yaml:174]: The same change to base.yaml in cconly directory.

**Review by wjun — APPROVED** [review]
